scripts/apple-silicon-hdf5-autogen.patch

```diff
@@ -0,0 +1,14 @@
+diff --git a/config/BuildSystem/config/packages/hdf5.py b/config/BuildSystem/config/packages/hdf5.py
+index f218d22bea..53451365aa 100644
+--- a/config/BuildSystem/config/packages/hdf5.py
++++ b/config/BuildSystem/config/packages/hdf5.py
+@@ -85,6 +85,9 @@ class Configure(config.package.GNUPackage):
+     self.addToArgs(args,'LIBS',self.libraries.toStringNoDupes(self.dlib))
+     return args
+
++  def preInstall(self):
++    self.executeShellCommand('HDF5_ACLOCAL=$(which aclocal) HDF5_AUTOHEADER=$(which autoheader) HDF5_AUTOMAKE=$(which automake) HDF5_AUTOCONF=$(which autoconf) HDF5_LIBTOOL=$(which libtool) HDF5_M4=$(which m4) ./autogen.sh',cwd=self.packageDir,log=self.log)
++
```

Comment:
I am thinking whether or not we should contribute this back to PETSc. In a long term, it might be a good idea to have PETSc to maintain this

Reply:
Could you explain a bit more why we need to do this? Because of arm?